src/utils/common.ts

```diff
@@ -283,3 +283,83 @@ export const removeHoveredClassFromMonth = (month: string) => {
     });
   }
 };
+
+export const syncUpdatedProjectFinancesWithStartYear = (finances: IProjectFinances, startYear: number) => {
+  let convertedFinances: IProjectFinances = {
+    year: startYear,
+    budgetProposalCurrentYearPlus0: "0.00",
+    budgetProposalCurrentYearPlus1: "0.00",
+    budgetProposalCurrentYearPlus2: "0.00",
+    preliminaryCurrentYearPlus3: "0.00",
+    preliminaryCurrentYearPlus4: "0.00",
+    preliminaryCurrentYearPlus5: "0.00",
+    preliminaryCurrentYearPlus6: "0.00",
+    preliminaryCurrentYearPlus7: "0.00",
+    preliminaryCurrentYearPlus8: "0.00",
+    preliminaryCurrentYearPlus9: "0.00",
+    preliminaryCurrentYearPlus10: "0.00"
+  }
+
+  const yearDifference = finances.year - startYear;
+
+  for (const key in finances) {
+    if (key != "year") {
+      const num =  parseInt(key.replace(/\D/g, ""));
+      const convertedNumber = num + yearDifference;
+      if (convertedNumber >= 0 && convertedNumber < 3) {
+        const convertedKey = ("budgetProposalCurrentYearPlus" + convertedNumber) as keyof IProjectFinances;
+        const updatedFinances = {
+          [convertedKey]: finances[key as keyof IProjectFinances]
+        };
+        convertedFinances = { ...convertedFinances, ...updatedFinances };
+      } else if (convertedNumber <= 10) {
+        const convertedKey = ("preliminaryCurrentYearPlus" + convertedNumber) as keyof IProjectFinances;
+        const updatedFinances = {
+          [convertedKey]: finances[key as keyof IProjectFinances]
+        };
+        convertedFinances = { ...convertedFinances, ...updatedFinances };
+      }
+    }
+  }
+  return convertedFinances;
+}
+
```

Comment:
The logic inside this function should also be commented.

Reply:
I refactored this and the function below so that they would be easier to read and naming of things would explain the code better. I also added comments for the functions explaining why they are needed and what they do. Is there still some parts inside the functions where the logic is not clear enough?

Reply:
They look good now 👍🏻